packages/gatsby/src/utils/webpack.config.js

```diff
@@ -212,25 +212,9 @@ module.exports = async (
               compiler.hooks.done.tapAsync(
                 `gatsby-webpack-stats-extractor`,
                 (stats, done) => {
-                  let assets = {}
-
-                  for (let chunkGroup of stats.compilation.chunkGroups) {
-                    if (chunkGroup.name) {
-                      let files = []
-                      for (let chunk of chunkGroup.chunks) {
-                        files.push(...chunk.files)
-                      }
-                      assets[chunkGroup.name] = files.filter(
-                        f => f.slice(-4) !== `.map`
-                      )
-                    }
-                  }
-
                   fs.writeFile(
                     path.join(`public`, `webpack.stats.json`),
-                    JSON.stringify({
-                      assetsByChunkName: assets,
-                    }),
+                    JSON.stringify(stats.toJson()),
```

Comment:
Since the webpack stats can be very big (megabytes), I don’t recommend writing the whole stats to the file. You can use `stats.toJson()` so that the `childAssets` are calculated and available, but we should make the stats file as small as possible.

What I recommend is this: Bring back the `assetsByChunkName` but instead of being an array of string (e.g. `{ app: ['app-hash.js'] }`, make it an array of link entries, e.g. `{ app: { assets: ['app-hash.js'], childAssets: ... } }`).